auth/lib/ui/home_page.dart

```diff
@@ -110,13 +113,24 @@ class _HomePageState extends State<HomePage> {
       _allCodes = codes;
       hasTrashedCodes = false;
       hasNonTrashedCodes = false;
+      hasNonFavouriteCodes = false;
+      hasFavouriteCodes = false;
+
       for (final c in _allCodes ?? []) {
         if (c.isTrashed) {
           hasTrashedCodes = true;
         } else {
           hasNonTrashedCodes = true;
         }
-        if (hasNonTrashedCodes && hasTrashedCodes) {
+        if (c.isPinned) {
+          hasFavouriteCodes = true;
+        } else {
+          hasNonFavouriteCodes = true;
+        }
+        if (hasTrashedCodes && hasNonTrashedCodes) {
+          break;
+        }
+        if (hasFavouriteCodes && hasNonFavouriteCodes) {
           break;
         }
```

Comment:
This should be if (hasTrashedCodes && hasNonTrashedCodes && hasFavouriteCodes && hasNonFavouriteCodes) {
break;
}